azurelinuxagent/ga/update.py

```diff
@@ -1069,6 +1056,19 @@ def __upgrade_agent_if_permitted(self):
                 upgrade_type == AgentUpgradeType.Normal and next_normal_time <= now):
             raise AgentUpgradeExitException(upgrade_message)
 
+    @staticmethod
+    def _ensure_no_error_json_files():
+        """
+        Older versions of the agent used to maintain an error.json file to blacklist agents if they ever failed to start.
+        We've since deprecated that behavior. Delete any error.json files that might still exist to keep a clean state.
+        """
+        for path in glob.glob(os.path.join(conf.get_lib_dir(), AGENT_DIR_GLOB, AGENT_ERROR_FILE)):
+            try:
+                os.remove(path)
+            except Exception:
+                # Make a best case effort to delete the the error.json file
```

Comment:
log a message?

Reply:
(actually, i would not even bother cleaning up those files, but if we consider this important enough then we should log the error)

Reply:
The only reason I added this is to make sure the daemon doesn't blacklist wrongfully due to some uncleaned state. 
And yes, I'll add a log